pkg/imagefilter/filter.go

```diff
@@ -89,13 +89,34 @@ func (fl filter) Matches(distro distro.Distro, arch distro.Arch, imgType distro.
 			m1 := term.pattern.Match(distro.Name())
 			m2 := term.pattern.Match(arch.Name())
 			m3 := term.pattern.Match(imgType.Name())
-			m = m && (m1 || m2 || m3)
+			// Also check image type aliases
+			m4 := false
+			if !m3 {
+				for _, alias := range imgType.Aliases() {
+					if term.pattern.Match(alias) {
+						m4 = true
+						break
+					}
+				}
+			}
```

Comment:
This is already duplicated twice in this file, and aliases are checked also elsewhere, AFAIK. Would it make sense to factor this out to a dedicated function that checks the aliases? Or make it an imageType method?

Reply:
I wonder if we could just have a dedicated function here for now and then if we need it in other places we could move it to an imageType method?

Reply:
Sure. The way aliases work in `architecture.GetImageType()` is quite different compared to what the `filter` package does.